source/channel.c

```diff
@@ -303,21 +303,17 @@ struct channel_shutdown_task_args {
 
 static int s_channel_shutdown(struct aws_channel *channel, int error_code, bool shutdown_immediately);
 
+static void s_on_shutdown_completion_task(struct aws_task *task, void *arg, enum aws_task_status status);
+
 static void s_shutdown_task(struct aws_channel_task *task, void *arg, enum aws_task_status status) {
 
     (void)task;
 
     struct shutdown_task *shutdown_task = arg;
-
+    struct aws_channel *channel = shutdown_task->channel;
+    int error_code = shutdown_task->error_code;
+    bool shutdown_immediately = shutdown_task->shutdown_immediately;
     if (status == AWS_TASK_STATUS_RUN_READY) {
```

Comment:
This `if (status == AWS_TASK_STATUS_RUN_READY) ` should be removed. This was a lurking bug in the existing code.

The other possible status for a task is "CANCELLED". This can happen if the event-loop is being torn down. In those cases, we still want to finish shutting down the channels on that event-loop.